app/controllers/Api2.scala

```diff
@@ -113,4 +112,20 @@ class Api2 @Inject() (implicit val previewDataStore: PreviewDataStore,
     }
     Ok
   }
+
+  def setActiveAsset(atomId: String) = APIHMACAuthAction { implicit req =>
+    req.body.asJson.map { json =>
+      try {
+        val videoId = (json \ "youtubeId").as[String]
+
+        val status: Unit = ActiveAssetCommand(atomId, videoId).process()
+
+        Ok("made asset " + videoId + " active in atom " + atomId)
```

Comment:
There are a few places where errors just return raw strings. If we're going to move to JSON it would be sensible to do it everywhere.